projects/cdk/directives/autofilled/autofilled.directive.ts

```diff
@@ -1,53 +1,41 @@
-import {Directive, ElementRef, Inject, Output} from '@angular/core';
-import {typedFromEvent} from '@taiga-ui/cdk/observables';
-import {TuiDirectiveStylesService} from '@taiga-ui/cdk/services';
-import {map, skip} from 'rxjs/operators';
+import {
+    Directive,
+    ElementRef,
+    EventEmitter,
+    HostBinding,
+    HostListener,
+    Inject,
+    Output,
+} from '@angular/core';
 
-const ON = 'tuiAutofillOn';
-const OFF = 'tuiAutofillOff';
-const STYLE = `
-@keyframes ${ON} {
-    from {
-        content: '1';
-    }
-
-    to {
-        content: '2';
-    }
-}
-
-@keyframes ${OFF} {
-    from {
-        content: '2';
-    }
-
-    to {
-        content: '1';
-    }
-}`;
-
-// @dynamic
 @Directive({
     selector: '[tuiAutofilledChange]',
-    host: {
-        class: 'tui-autofill',
-    },
+    host: {class: 'tui-autofill'},
 })
 export class TuiAutofilledDirective {
+    @HostBinding('class._autofilled')
+    autofilled = false;
+
     @Output()
-    readonly tuiAutofilledChange = typedFromEvent(
-        this.elementRef.nativeElement,
-        'animationstart',
-    ).pipe(
-        map(({animationName}) => animationName === ON),
-        skip(1),
-    );
+    readonly tuiAutofilledChange = new EventEmitter<boolean>();
+
+    constructor(@Inject(ElementRef) private readonly elementRef: ElementRef<Element>) {}
+
+    @HostListener('transitionstart', ['$event'])
+    transitionStartHandler({propertyName, target}: TransitionEvent): void {
+        if (propertyName.includes('box-shadow') && target === this.inputElement) {
```

Comment:
Let's just check that `target.matches('input')`, it would be way simpler and not much less safe since you just reach for the closest `input` tag anyway.

Reply:
fixed